### PR TITLE
support grpc-v2 for torchserve runtime

### PR DESCRIPTION
**Pull request by jagadeeshi2i**
Depends on https://github.com/pytorch/serve/pull/2566

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevent result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -354,6 +354,7 @@ const (
 const (
 	ModelClassLabel = "modelClass"
 	ServiceEnvelope = "serviceEnvelope"
+	ISVCProtocol    = "iscvProtocol"
```
Comment (yuzisun) [inline pkg/constants/constants.go:358]:
```suggestion
	ISVCProtocol    = "protocolVersion"
```

**Comment by yuzisun** [comment]
@jagadeeshi2i Can you help add a test for gRPC torchserve ?

File: test/e2e/predictor/test_torchserve.py [hunk 2]
```diff
@@ -39,34 +38,33 @@ def test_torchserve_kserve():
     predictor = V1beta1PredictorSpec(
         min_replicas=1,
         pytorch=V1beta1TorchServeSpec(
+            image="pytorch/torchserve-kfs:0.8.2",
```
Comment (yuzisun) [inline test/e2e/predictor/test_torchserve.py:41]: Do we still need this override as it is already configured in the kustomize.yaml ?

File: config/runtimes/kustomization.yaml [hunk 3]
```diff
@@ -2,50 +2,50 @@ apiVersion: kustomize.config.k8s.io/v1beta1
...
-- kserve-torchserve.yaml
+  - kserve-tensorflow-serving.yaml
+  - kserve-sklearnserver.yaml
+  - kserve-mlserver.yaml
+  - kserve-xgbserver.yaml
+  - kserve-tritonserver.yaml
+  - kserve-pmmlserver.yaml
+  - kserve-paddleserver.yaml
+  - kserve-lgbserver.yaml
+  - kserve-torchserve.yaml
```
Comment (yuzisun) [inline config/runtimes/kustomization.yaml:13]: Are these indentation expected ?

File: pkg/constants/constants.go (hunk 1 above)
Comment (yuzisun) [reply pkg/constants/constants.go:358]: also how is this label used?

**Comment by yuzisun** [comment]
@jagadeeshi2i Can you help update the PR ?

**Comment by andyi2it** [comment]
/close
Closing the PR as it is incorporated in other PRs

**Pull request closed without merging**